### PR TITLE
[#208] Moved existing samples into single sample app

### PR DESCRIPTION
**Pull request by LachlanMcKee**
## Description
Moved existing samples into single sample app.

I have recorded two videos demonstrating the behaviour. Note that I have disabled animations on my emulator (they are not broken)

**Sample app first use**
[samples-app-first-use.webm](https://user-images.githubusercontent.com/1333155/195810250-763df557-8d90-472c-9408-cd1869454e30.webm)

**Sample app second use**
[samples-app-second-use.webm](https://user-images.githubusercontent.com/1333155/195810279-a9a168e8-0a61-4ba6-9831-366f9461d1ae.webm)

We can potentially improve the UI appearance in a follow-up ticket? (as it could be more interesting)

## Check list

- [x] I have updated `CHANGELOG.md` if required.
- [x] I have updated documentation if required.

### CODE REVIEW

File: samples/app/src/main/kotlin/com/bumble/appyx/app/MainActivity.kt [hunk 1]
```diff
@@ -24,14 +26,18 @@ class MainActivity : NodeActivity() {
...
         setContent {
-            AppyxSampleAppTheme {
-                // A surface container using the 'background' color from the theme
-                Surface(color = MaterialTheme.colors.background) {
-                    Column {
-                        NodeHost(integrationPoint = appyxIntegrationPoint) {
-                            RootNode(
-                                buildContext = it,
-                            )
+            CompositionLocalProvider(
```
Comment (LachlanMcKee) [inline samples/app/src/main/kotlin/com/bumble/appyx/app/MainActivity.kt]: Needed for the `navigation-compose` sample

Comment (zsoltk) [reply samples/app/src/main/kotlin/com/bumble/appyx/app/MainActivity.kt]: I'm a bit unhappy about it as it gives the impression it's needed in general, and people might even copy-paste it needlessly.

Comment (LachlanMcKee) [reply samples/app/src/main/kotlin/com/bumble/appyx/app/MainActivity.kt]: I'll see what I can do about it

Comment (LachlanMcKee) [reply samples/app/src/main/kotlin/com/bumble/appyx/app/MainActivity.kt]: @zsoltk done. I moved the `LocalIntegrationPoint` to the point at which it is needed by `ComposeNavigationRoot`.

The entire point of that sample is how to use Appyx within an existing Jetpack compose project, so I do not want to expose a root Appyx node from that sample

**Review by zsoltk — APPROVED** [review]

**Pull request merged**